packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -299,9 +303,47 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
   }
 
   private void handleChoosePictureResult(int resultCode, Intent data) {
-    if (resultCode == Activity.RESULT_OK && data != null) {
+    if (resultCode == Activity.RESULT_OK && data != null && data.getData() != null) {
+      //Extract the file path from the uri.
       String path = fileUtils.getPathFromUri(activity, data.getData());
-      handleResult(path);
+      //Path may be null if, for example, source is remote.
+      if (path == null) {
+        InputStream inputStream = null;
```

Comment:
Just a drive-by comment: I would at least extract this logic into a separate method (called `getRemotePathFromUri()` for example) to make the code a bit clearer. 

Maybe even move this logic to `FileUtils` class if we go crazy. If the `Uri` has a distinctive `scheme` or `authority`, it could be added as an `else if` at [Line 90 in the FileUtils class](https://github.com/flutter/plugins/blob/master/packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java#L90).

Something like:

**FileUtils.java (starting from line 90)**
```java
...
    } else if (isRemoteFile(uri)) {
        // Move the logic for getting remote path from the uri here.
        // Then return the resolved path.
    } 
...
```

Disclaimer: I'm not in the Flutter team and you can just ignore this feedback if you want. :)

Reply:
Problem is the authority is not distintive.
For example local file from google photos has the same authority then remote file. 
However the local file path is correctly retrieved but not the remote. 
That's why I used the logic as a fallback cause this always works but need to create a temp file... Hence the idea was to say: let's try to retrieve the path in every manner we may know of and fallback to inputstream and tempFile if those methods fail. It allows to keep the plugin to work even if a provider change its distintive authority